docs/providers/aws/guide/iam.md

```diff
@@ -0,0 +1,279 @@
+<!--
+title: Serverless Framework - AWS Lambda Guide - IAM
+menuText: IAM
+menuOrder: 12
+description: How to set up the different roles on a service and function level
+layout: Doc
+-->
+
+<!-- DOCS-SITE-LINK:START automatically generated  -->
+### [Read this on the main serverless docs site](https://www.serverless.com/framework/docs/providers/aws/guide/iam)
+<!-- DOCS-SITE-LINK:END -->
+
+# Defining IAM Rights
+
+Serverless provides no-configuration rights provisioning by default.
+However you can always define roles on a service or function level if you need to.
+
+## Default Role Management
+
+The default rights provisioning approach requires no configuration and defines a role that is shared by all of the Lambda functions in your service. A policy is also created and is attached to the generated role. Any additional specific rights are added to the role by defining provider level `iamRoleStatements` that will be merged into the generated policy.
+
+### Adding Custom IAM Role Statements to the Default Policy
+
+By default, your Lambda functions will be provided with the right to create and write to CloudWatch logs. Further, if you have specified VPC security groups and subnets for your lambdas to use then the EC2 rights necessary to attach to the VPC via an ENI will be added into the default IAM policy.
+
+If you want to give permission to your functions to access certain resources on your AWS account, you can add custom IAM role statements to your service by adding the statements in the `iamRoleStatements` array in the `provider` object. As those statements will be merged into the CloudFormation template you can use `Join`, `Ref` or any other CloudFormation method or feature. You're also able to either use YAML for defining the statement (including the methods) or use embedded JSON if you prefer it. Here's an example that uses all of these:
+
+```yml
+service: new-service
+
+provider:
+  name: aws
+  iamRoleStatements:
+    -  Effect: "Allow"
+       Action:
+         - "s3:ListBucket"
+       Resource: { "Fn::Join" : ["", ["arn:aws:s3:::", { "Ref" : "ServerlessDeploymentBucket"} ] ] }
```

Comment:
@pmuens I know this is valid yaml and probably not related to this PR, but imo the examples should be in yaml syntax for consistency and simplicity. So I'd ditch the brackets, quotes...etc